service/src/main/java/bio/terra/workspace/app/StartupInitializer.java

```diff
@@ -1,64 +1,49 @@
 package bio.terra.workspace.app;
 
-import bio.terra.common.db.DataSourceInitializer;
+import bio.terra.common.db.DataSourceManager;
 import bio.terra.common.migrate.LiquibaseMigrator;
 import bio.terra.landingzone.library.LandingZoneMain;
 import bio.terra.workspace.app.configuration.external.FeatureConfiguration;
-import bio.terra.workspace.app.configuration.external.StartupConfiguration;
 import bio.terra.workspace.app.configuration.external.WorkspaceDatabaseConfiguration;
-import bio.terra.workspace.db.WorkspaceDao;
-import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.job.StairwayInitializerService;
 import bio.terra.workspace.service.workspace.WsmApplicationService;
 import javax.sql.DataSource;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import org.springframework.context.ApplicationContext;
 
 public final class StartupInitializer {
-  private static final Logger logger = LoggerFactory.getLogger(StartupInitializer.class);
   private static final String changelogPath = "db/changelog.xml";
 
   public static void initialize(ApplicationContext applicationContext) {
     // Initialize or upgrade the database depending on the configuration
-    LiquibaseMigrator migrateService = applicationContext.getBean(LiquibaseMigrator.class);
+    DataSourceManager dataSourceManager = applicationContext.getBean(DataSourceManager.class);
     WorkspaceDatabaseConfiguration workspaceDatabaseConfiguration =
         applicationContext.getBean(WorkspaceDatabaseConfiguration.class);
-    JobService jobService = applicationContext.getBean(JobService.class);
+    LiquibaseMigrator migrateService = applicationContext.getBean(LiquibaseMigrator.class);
+    StairwayInitializerService stairwayInitializerService =
+        applicationContext.getBean(StairwayInitializerService.class);
     WsmApplicationService appService = applicationContext.getBean(WsmApplicationService.class);
     FeatureConfiguration featureConfiguration =
         applicationContext.getBean(FeatureConfiguration.class);
-    StartupConfiguration startupConfiguration =
-        applicationContext.getBean(StartupConfiguration.class);
 
     // Log the state of the feature flags
     featureConfiguration.logFeatures();
 
     // Migrate the database
     DataSource workspaceDataSource =
-        DataSourceInitializer.initializeDataSource(workspaceDatabaseConfiguration);
```

Comment:
`DataSourceInitializer` docstring says:
```
Prefer using @DataSourceManager instead of this code so that connection pools are closed when the Spring application context is deleted.
```

Which also has the benefit of making the code easier to test.